### PR TITLE
Clean up FakeRedis

### PR DESCRIPTION
**Pull request by keeferrourke**
Follow-on from https://github.com/cashapp/misk/pull/2685

1. Remove lock object and `synchronized{}` calls
2. Mark all public methods as `@Synchronized`
3. Fix style and formatting nits

### CODE REVIEW

File: misk-redis/src/main/kotlin/misk/redis/FakeRedis.kt [hunk 1]
```diff
@@ -19,193 +19,156 @@ class FakeRedis : Redis {
...
+  @Synchronized
   override fun mset(vararg keyValues: ByteString) {
-    synchronized(lock) {
-      require(keyValues.size % 2 == 0) { "Wrong number of arguments to mset" }
+    require(keyValues.size % 2 == 0) { "Wrong number of arguments to mset" }
 
-      (0 until keyValues.size step 2).forEach {
-        set(keyValues[it].utf8(), keyValues[it + 1])
-      }
+    (keyValues.indices step 2).forEach {
```
Comment (keeferrourke) [inline misk-redis/src/main/kotlin/misk/redis/FakeRedis.kt:53]: IntelliJ suggestion: `.indices` is equivalent to `0 until iterable.size`

File: misk-redis/src/main/kotlin/misk/redis/FakeRedis.kt [hunk 2]
```diff
@@ -19,193 +19,156 @@ class FakeRedis : Redis {
...
-      return keyValueStore.remove(key) != null
+    if (!keyValueStore.containsKey(key)) {
+      return false
     }
+    return keyValueStore.remove(key) != null
   }
 
-  override fun del(vararg keys: String): Int {
-    synchronized(lock) {
-      // Call delete on each key and count how many were successful
```
Comment (keeferrourke) [inline misk-redis/src/main/kotlin/misk/redis/FakeRedis.kt:50]: redundant comment

File: misk-redis/src/main/kotlin/misk/redis/FakeRedis.kt [hunk 3]
```diff
@@ -19,193 +19,156 @@ class FakeRedis : Redis {
...
-      keyValueStore.putIfAbsent(key, Value(
-        data = value,
-        expiryInstant = clock.instant().plusSeconds(expiryDuration.seconds)
-      )) == null
-    }
+    return setWithExpiry(key, value, clock.instant().plusSeconds(expiryDuration.seconds))
+  }
+
+  private fun setWithExpiry(key: String, value: ByteString, expiryInstant: Instant): Boolean {
+    return keyValueStore.putIfAbsent(key, Value(data = value, expiryInstant)) == null
```
Comment (keeferrourke) [inline misk-redis/src/main/kotlin/misk/redis/FakeRedis.kt:168]: Pulled out this common implementation of `setnx`

**Review by tso — APPROVED** [review]
nice

**Pull request merged**